### PR TITLE
Add build artifacts to release

### PR DESCRIPTION
**Pull request by BSFishy**
### Description
Adds build artifacts to the release drafter workflow. By running `yarn build` before drafting the release, we can make sure releases include all of the compiled JS and CSS needed by consumers
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] All tests pass
  - [x] `yarn lint`
  - [x] `yarn test-unit`
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/oui/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by joshuarrrr — DISMISSED** [review]

File: .github/workflows/release-drafter.yml [hunk 1]
```diff
@@ -13,8 +13,24 @@ jobs:
...
+          registry-url: 'https://registry.npmjs.org'
+      - name: Setup Yarn
+        run: |
+          npm uninstall -g yarn
+          npm i -g yarn@1.22.10
+          yarn install --frozen-lockfile
+      - name: Generate build
+        run: |
+          yarn build
+          yarn pack --filename oui.tar.gz
```
Comment (gaiksaya) [inline .github/workflows/release-drafter.yml:29]: Recently we release reporting0cli and they are creating the tarball using npm pack. See: https://github.com/opensearch-project/reporting-cli/blob/main/.github/workflows/release-drafter.yml#L17-L29
is that different from whet you are doing here?

File: .github/workflows/release-drafter.yml [hunk 2]
```diff
@@ -13,8 +13,24 @@ jobs:
...
       - name: Release
         uses: softprops/action-gh-release@v1
         with:
           draft: true
-          generate_release_notes: true
\ No newline at end of file
+          generate_release_notes: true
+          files: |
+            oui.tar.gz
```
Comment (gaiksaya) [inline .github/workflows/release-drafter.yml:36]: The current pipeline unzipping whatever artifacts are inside a tarball. You might want to zip this inside one more tarball.

**Review by gaiksaya — COMMENTED** [review]
I don't have much context on how artifacts should be packed for this project. Added some comments based on what other project did.

File: .github/workflows/release-drafter.yml (hunk 1 above)
Comment (BSFishy) [reply .github/workflows/release-drafter.yml:29]: I don't think so. I think `yarn pack` passes through or emulates the functionality of `npm pack` while still being a yarn command. I think I'd prefer to stick with yarn commands as much as possible so we don't have any chance of worrying about conflicting `yarn.lock` and `package-lock.json` files

File: .github/workflows/release-drafter.yml (hunk 2 above)
Comment (BSFishy) [reply .github/workflows/release-drafter.yml:36]: For clarification, you're saying that `oui.tar.gz` should be wrapped inside of another tarball?

Comment (gaiksaya) [reply .github/workflows/release-drafter.yml:36]: Yes

Comment (BSFishy) [reply .github/workflows/release-drafter.yml]: Done

**Review by joshuarrrr — APPROVED** [review]

**Review by ashwin-pc — APPROVED** [review]

**Review by gaiksaya — APPROVED** [review]

**Pull request merged**